project/Doc.scala

```diff
@@ -82,3 +74,63 @@ object Unidoc {
       file
   }
 }
+
+/**
+ * For projects with few (one) classes there might not be any diagrams.
+ */
+object ScaladocNoVerificationOfDiagrams extends AutoPlugin {
+
+  override def trigger = noTrigger
+  override def requires = Scaladoc
+
+  override lazy val projectSettings = Seq(
+    Scaladoc.validateDiagrams in Compile := false
+  )
+}
+
+/**
+ * Unidoc settings for root project. Adds unidoc command.
+ */
+object UnidocRoot extends AutoPlugin {
+
+  object CliOptions {
+    val genjavadocEnabled = CliOption("akka.genjavadoc.enabled", false)
+  }
+
+  override def trigger = allRequirements
+  override def requires = RootSettings
+
+  def settings(ignoreAggregates: Seq[Project], ignoreProjects: Seq[Project]) = {
+    val withoutAggregates = ignoreAggregates.foldLeft(inAnyProject) { _ -- inAggregates(_, transitive = true, includeRoot = true) }
+    val docProjectFilter = ignoreProjects.foldLeft(withoutAggregates) { _ -- inProjects(_) }
+
+    inTask(unidoc)(Seq(
+      unidocProjectFilter in ScalaUnidoc := docProjectFilter,
+      unidocProjectFilter in JavaUnidoc := docProjectFilter,
+      apiMappings in ScalaUnidoc := (apiMappings in (Compile, doc)).value
+    ))
+  }
+
+  override lazy val projectSettings =
+    CliOptions.genjavadocEnabled.ifTrue(scalaJavaUnidocSettings).getOrElse(scalaUnidocSettings) ++
+    settings(Seq(Project(base=file("akka-samples"), id="akka-samples")), Seq(
+      Project(base=file("akka-remote-tests"), id="akka-remote-tests"),
+      Project(base=file("akka-jmh"), id="akka-jmh")
```

Comment:
This looks a bit fragile, why not leave the selection of what to exclude in AkkaBuild where it is statically verified?